### ISSUE TITLE
Support Intellij IDEA

### ISSUE THREAD

**Issue by minhnguyenvan95** [RicardoNiepel/C4-PlantUML#63]
Do you have plant to support intellij idea as a plugin ?

**Comment by stawirej** [comment]
in the meantime you can check https://github.com/stawirej/c4-intellij-live-template

**Comment by minhnguyenvan95** [comment]
Thanks @stawirej , just found this plugins 

- Intellij IDEA Plugins: https://plugins.jetbrains.com/plugin/7017-plantuml-integration

**Comment by H3llK33p3r** [comment]
Hi, am I the only one that the import of the Intellij live templates is not working ? After the restart I haven't the templates and cannot find in my IDE settings. Moreover, why the template are in other section and not plantuml section (reading the xml in the zip) ?

**Comment by Potherca** [comment]
@H3llK33p3r Apologies for my slow response, I monitor comments less strictly than (new) issues.

Regarding your problem, I tried importing the live templates (zip file) and failed. After some researching, I found the source for the failure in my setup. 

According to the [Jetbrains documentation about sharing live templates](https://www.jetbrains.com/help/idea/sharing-live-templates.html), what happens when the contents of the zip file (`c4.xml`) is imported, is that the XML file is placed into the `templates` directory of the [IDE configuration directory](https://www.jetbrains.com/help/idea/directories-used-by-the-ide-to-store-settings-caches-plugins-and-logs.html#config-directory).

In my case this was `~/.config/JetBrains/PhpStorm2022.2/templates/c4.xml`. However, as I have [IDE Settings Sync](https://www.jetbrains.com/help/idea/sharing-your-ide-settings.html#IDE_settings_sync) enabled, the IDE looks for live template files under the `jba_config/templates` directory. In my case `~/.config/JetBrains/PhpStorm2022.2/jba_config/templates/c4.xml`.

Once I moved the XML file form `templates/` to `jba_config/templates/` (and rebooted my IDE) everything worked as expected.

If you have IDE Settings Sync enabled, this workaround should work for you, otherwise more debugging will be required.

I don't know if this is a known issue and/or if Jetbrains is aware of this, but this looks like a bug in Intellij to me.

**Comment by H3llK33p3r** [comment]
Hi, I'm also sorry for my slow response. When I saw your answer I checked and it cannot work for me (I'm also using a git sync settings). 

Because at this time I don't need to work on schema, I delay the checking of why it's not working. Your information was great because like you, it seems the templates directory is not check and my configure are stored into : 

`...AppData\Roaming\JetBrains\IntelliJIdea2022.2\settingsRepository\repository\templates` (on windows). Puttting the file into this directory and it's working.
